### PR TITLE
Adding Consensus Profiles + Code

### PR DESCRIPTION
**Pull request by gwaygenomics**
This PR adds 5 files and addresses #36 

The files include:

1. `README.md` describing the consensus details and procedure
2. Notebook performing the procedure
3. nbconverted `.py` file corresponding to the notebook
4. median consensus profiles
5. MODZ consensus profiles

A couple of decision points to note:

* I did not apply a feature selection procedure (step 5 in #36).  My rationale is b/c feature selection is fairly experiment-specific, can be done quickly in different experiments, and to reduce the number of files. Happy to add if there are other opinions!
* I created per-well DMSO consensus signatures. This was recommended previously to assess plate-level batch effects.

### CODE REVIEW

**Comment by gwaygenomics** [comment]
cc @shntnu

**Comment by shntnu** [comment]
> * I did not apply a feature selection procedure (step 5 in #36).  My rationale is b/c feature selection is fairly experiment-specific, can be done quickly in different experiments, and to reduce the number of files. Happy to add if there are other opinions!

I agree; more here https://github.com/broadinstitute/lincs-cell-painting/issues/36#issuecomment-629426676

**Comment by shntnu** [comment]
Is it reasonable to structure it like this? 
```
…
├── profiles
│   └── 2016_04_01_a549_48hr_batch1
│       └── SQ00015167
│           ├── SQ00015167.csv
│           ├── SQ00015167_augmented.csv
│           ├── SQ00015167_normalized.csv
│           └── SQ00015167_normalized_feature_select.csv
├── consensus
│   └── 2016_04_01_a549_48hr_batch1
│       ├── 2016_04_01_a549_48hr_batch1_consensus_modz.csv.gz
│       └── 2016_04_01_a549_48hr_batch1_consensus_median.csv.gz
…
```

File: profiles/consensus/scripts/nbconverted/build-consensus-signatures.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    consensus_df = (
+        pd.concat([dmso_df, compound_df], axis="rows", sort=True)
+        .reset_index(drop=True)
+        .loc[:, dmso_replicate_cols + cp_features]
+    )
+
+    print(operation)
+    print(consensus_df.shape)
+
+    consensus_df.to_csv(consensus_file, sep=",", compression="gzip", float_format="%5g")
```
Comment (shntnu) [inline profiles/consensus/scripts/nbconverted/build-consensus-signatures.py:225]: `, index=False`

File: profiles/consensus/scripts/nbconverted/build-consensus-signatures.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+# ### b) Generate consensus profiles per compound-dose pair
+
+# In[12]:
+
+
+compound_replicate_cols = [
+    "Metadata_broad_sample",
+    "Metadata_mmoles_per_liter",
+    "Metadata_dose_recode",
```
Comment (shntnu) [inline profiles/consensus/scripts/nbconverted/build-consensus-signatures.py:180]: We should also include `Metadata_Plate_Map_Name` here. This makes is easier to detect batch effects with the small handful (n=2 I think) of compounds present on multiple platemaps.

Same thing with `dmso_replicate_cols`. You will get 24 wells x 28 plate maps = 672 DMSO profiles, which is what we want.

File: profiles/consensus/scripts/nbconverted/build-consensus-signatures.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+
+# ## Create Consensus Profiles
+# 
+# ### a) Generate consensus profiles for DMSO by well
+
+# In[10]:
+
+
+dmso_replicate_cols = [
```
Comment (shntnu) [inline profiles/consensus/scripts/nbconverted/build-consensus-signatures.py:142]: See https://github.com/broadinstitute/lincs-cell-painting/pull/40/files#r426948458

File: profiles/consensus/scripts/nbconverted/build-consensus-signatures.py (hunk 2 above)
Comment (gwaygenomics) [reply profiles/consensus/scripts/nbconverted/build-consensus-signatures.py]: this is great!! Thanks for catching

**Comment by gwaygenomics** [comment]
Thanks for the comments @shntnu - the structure and code is much improved. Ready for your re-review when you get a chance

File: consensus/scripts/nbconverted/build-consensus-signatures.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    )
+
+
+# ## Merge and Output Consensus Signatures
+
+# In[14]:
+
+
+for operation in operations:
+    consensus_file = f"{operation}_consensus_signatures_level5.csv.gz"
```
Comment (shntnu) [inline consensus/scripts/nbconverted/build-consensus-signatures.py:217]: Do you prefer this to the suggested name here https://github.com/broadinstitute/lincs-cell-painting/pull/40#issuecomment-630480994 (use batch name, similar to the way you'd use plate name)

Comment (gwaygenomics) [reply consensus/scripts/nbconverted/build-consensus-signatures.py:217]: oh yes, overlooked that, my fault - will update in next commit!

**Comment by gwaygenomics** [comment]
> Have a look at #40 (comment). The rest looks 💯 !

all set! Going to merge

**Comment by shntnu** [comment]
> all set! Going to merge

Hooray!

**Pull request merged**